app/models/post.rb

```diff
@@ -0,0 +1,15 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments
+  has_many :likes
+
+  # Custom method: Updates the posts counter for a user
+  def update_user_posts_counter
+    user.update(post_counter: user.posts.count)
+  end
+
+  # Custom method: Returns the 5 most recent comments for a given post
+  def recent_comments(limit = 5)
+    comments.order(created_at: :desc).limit(limit)
+  end
```

Comment:
- [ ] [Optional] It would be better professionally if you use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) to be specific the `after_save` method to call the `update_user_posts_counter` method every time a new post has been saved to the database, then make the `update_user_posts_counte` method private.
```suggestion
 after_save :update_user_posts_counter

 # Custom method: Returns the 5 most recent comments for a given post
  def recent_comments(limit = 5)
    comments.order(created_at: :desc).limit(limit)
  end
  
 private
 # Custom method: Updates the posts counter for a user
  def update_user_posts_counter
    user.update(post_counter: user.posts.count)
  end
```